src/shared/libfido2-util.c

```diff
@@ -830,7 +843,17 @@ int fido2_generate_hmac_hash(
                            emoji_enabled() ? special_glyph(SPECIAL_GLYPH_TOUCH) : "",
                            emoji_enabled() ? " " : "");
 
-        r = sym_fido_dev_make_cred(d, c, NULL);
+        /* If we are using the user PIN, then we must pass that PIN to the get_assertion call below, or
+         * the authenticator will use the non-user-verification HMAC secret (which differs from the one when
+         * the PIN is passed).
+         *
+         * Rather than potentially trying and failing to create the credential, just collect the PIN first
+         * and then pass it to both the make_credential and the get_assertion operations. */
+        if (FLAGS_SET(lock_with, FIDO2ENROLL_PIN))
+            r = FIDO_ERR_PIN_REQUIRED;
+        else
+            r = sym_fido_dev_make_cred(d, c, NULL);
```

Comment:
8 space indentation please, i.e follow coding style

Reply:
still not fixed

Reply:
I hadn't pushed the extra spaces on these two lines yet. It's fixed now.